docs/usage/shoot_pod_autoscaling_best_practices.md

```diff
@@ -0,0 +1,327 @@
+- As for minimum and maximum, let your high availability requirements determine the minimum and your theoretical maximum load determine the maximum, flanked with alerts to detect erroneous run-away out-scaling or the actual nearing of your practical maximum load, so that you can intervene.
+- Vertical Pod Autoscaling (VPA): Use always for containers that have a significant usage (e.g. any container above 50m CPU or 100M memory) and a significant spread over time (by more than 2x), i.e. ignore small (e.g. side-cars) or static (e.g. Java statically allocated heap) containers, but otherwise use it to provide the resources needed on the one hand and keep the costs in check on the other hand.
+- As a rule of thumb, set the initial requests to the 5th percentile of the actually observed CPU resp. memory usage in production. Since VPA may need some time at first to respond and evict pods, this is especially critical early after deployment. The lower bound, below which pods will be immediately evicted, converges much faster than the upper bound, above which pods will be immediately evicted, but it isn't instant, e.g. after 5 minutes the lower bound is just at 60% of the computed lower bound; after 12 hours the upper bound is still at 300% of the computed upper bound (see [code](https://github.com/kubernetes/autoscaler/blob/b3a501cbe11e46bea1f8879d39c8436ef03e7139/vertical-pod-autoscaler/pkg/recommender/logic/recommender.go#L118-L143)). Unlike with HPA, you don't need to be as cautious when updating the higher-level resource in the case of VPA. As long as VPA's mutating webhook (VPA Admission Controller) is operational (which also the VPA Updater checks before evicting pods), it's generally safe to update the higher-level resource. However, if you want to err on the side of caution, it's advisable to check the health of the VPA Admission Controller first. If it's not up and running, any new pods that are spawned (e.g. as a consequence of a rolling update of the higher-level resource or for any other reason) will not be mutated. Instead, they will receive whatever requests are currently configured at the higher-level resource, which can lead to catastrophic resource under-reservation. Gardener deploys the VPA Admission Controller in HA - if unhealthy, it is reported under the `ControlPlaneHealthy` shoot status condition.
```

Comment:
For initial requests of Gardener components, recently I have been aiming at a setup which is meant to start at an idle level consumption, and be actively scaled by VPA, as load arrives. Not at one which has a chance of surviving in the absence of VPA. For an idle-level setup, the instructions would be in the lines of:

Run your pods integrated into an idle instance of whatever system they are part of. Let the system sit idle for another 30 minutes. Take the 90th percentile of the pod's observed resource consumption during those idle 30 minutes, multiply it by factor 1.15, and use that value as DefaultRequests.

Both approaches yield different values and I think each has cases where it works better. I think the one you propose works better when the large majority of application instances are under a relatively similar load. For Gardener components, where load varies by orders of magnitude, I have been aiming at starting at a minimal stable ground state and then relying entirely on VPA.

Reply:
But we know from painful outages that this approach, starting fairly low, takes longer to resolve (e.g. many OOM bump-up loops).